pyacq/core/tests/test_tools.py

```diff
@@ -0,0 +1,70 @@
+from pyacq.core.stream  import OutputStream, InputStream
+from pyacq.core.tools  import ThreadPollInput
+from pyqtgraph.Qt import QtCore, QtGui
+import pyqtgraph as pg
+
+import numpy as np
+import weakref
+import time
+
+
+def test_ThreadPollInput():
+    app = pg.mkQApp()
+    
+    nb_channel = 16
+    chunksize = 1024
+    stream_spec = dict(protocol = 'tcp', interface = '127.0.0.1', port='*', 
+                       transfermode = 'plaindata', streamtype = 'analogsignal',
+                       dtype = 'float32', shape = (-1, nb_channel), compression ='',
+                       scale = None, offset = None, units = '', timeaxis = 0)
+    
+    class ThreadSender(QtCore.QThread):
+        def __init__(self, output_stream, parent = None):
+            QtCore.QThread.__init__(self, parent)
+            self.output_stream = weakref.ref(output_stream)
```

Comment:
To avoid that a QThread have a reference to the Stream that belong to the Node.
This avoid loop reference.
Node.thread
Node.input
thread.input << weakref here.

Furthermore Stream IS a zmq socket and if the Node disapear (with its input_stream) then no worry in the thread.

Are you agree ?